<a href="https://colab.research.google.com/github/sksuman97/WorldBank-Global-Education-Analysis/blob/main/Sumanta_Muduli_World_Bank_Global_Education_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The World Bank EdStats All Indicator Query holds over 4,000 internationally comparable indicators that describe education access, progression, completion, literacy, teachers, population, and expenditures.The indicators cover the education cycle from pre-primary to vocational and tertiary education and also holds learning outcome data from international and regional learning assessments (e.g. PISA, TIMSS, PIRLS), equity data from household surveys, and projection/attainment data. </b>

## <b> Explore and analyze the data to identify variation of indicators across the globe, which countries are more alike and different. Feel free to add more extensive analyses and details.</b>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [65]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [66]:
country_series=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Cohort Nilgiri/Module 1/capstone project/data/EdStatsCountry-Series.csv')

In [67]:
country=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Cohort Nilgiri/Module 1/capstone project/data/EdStatsCountry.csv')

In [68]:
stats_data=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Cohort Nilgiri/Module 1/capstone project/data/EdStatsData.csv')

In [69]:
footnote=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Cohort Nilgiri/Module 1/capstone project/data/EdStatsFootNote.csv')

In [70]:
stats_series=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Cohort Nilgiri/Module 1/capstone project/data/EdStatsSeries.csv')

In [71]:
country_series.head()

CountryCode  ... Unnamed: 3
0         ABW  ...        NaN
1         ABW  ...        NaN
2         AFG  ...        NaN
3         AFG  ...        NaN
4         AFG  ...        NaN

[5 rows x 4 columns]

In [72]:
country_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


In [73]:
# Drop the column unnamed as it does not have any non null values
del country_series['Unnamed: 3']

In [74]:
country_series.head()

CountryCode  ...                                        DESCRIPTION
0         ABW  ...  Data sources : United Nations World Population...
1         ABW  ...  Data sources: United Nations World Population ...
2         AFG  ...  Data sources: United Nations World Population ...
3         AFG  ...                 Estimates are based on regression.
4         AFG  ...  Data sources : United Nations World Population...

[5 rows x 3 columns]

In [75]:
country_series['SeriesCode'].unique()

array(['SP.POP.TOTL', 'SP.POP.GROW', 'NY.GDP.PCAP.PP.CD',
       'NY.GDP.MKTP.PP.KD', 'NY.GNP.MKTP.PP.CD', 'NY.GDP.MKTP.PP.CD',
       'NY.GDP.PCAP.PP.KD', 'NY.GNP.PCAP.PP.CD', 'SP.POP.1564.TO.ZS',
       'SP.POP.TOTL.MA.ZS', 'SP.POP.TOTL.FE.ZS', 'SP.POP.0014.TO.ZS',
       'NY.GNP.PCAP.CD', 'NY.GDP.PCAP.CD', 'NY.GDP.PCAP.KD',
       'SP.POP.1564.MA.IN', 'SP.POP.0014.TO', 'SP.POP.1564.TO',
       'SP.POP.1564.FE.IN', 'SP.POP.0014.MA.IN', 'SP.POP.0014.FE.IN'],
      dtype=object)

In [76]:
country_series['CountryCode'].unique().shape

(211,)

In [77]:
country_series['DESCRIPTION'].unique()

array(['Data sources : United Nations World Population Prospects',
       'Data sources: United Nations World Population Prospects',
       'Estimates are based on regression.',
       'Data sources : Institute of Statistics, Eurostat',
       'Data sources : Australian Bureau of Statistics',
       'Including Other Territories comprising Jervis Bay Territory, Christmas Island and the Cocos (Keeling) Islands. ',
       'Data sources: Australian Bureau of Statistics',
       'Data sources : Eurostat',
       'Data sources : Eurostat, State Statistical Committee, United Nations World Population Prospects',
       'Data sources: Eurostat, State Statistical Committee, United Nations World Population Prospects',
       'Data sources : National Statistical Committee of the Republic of Belarus',
       'Data sources: National Statistical Committee of the Republic of Belarus',
       'Data sources : Bermuda Government Department of Statistics',
       'Data sources: Bermuda Government Departme

In [78]:
country_series[['SeriesCode','DESCRIPTION']].iloc[40:100]

SeriesCode                                        DESCRIPTION
40        SP.POP.TOTL  Data sources : Eurostat, State Statistical Com...
41        SP.POP.GROW  Data sources: Eurostat, State Statistical Comm...
42        SP.POP.GROW  Data sources: United Nations World Population ...
43        SP.POP.TOTL  Data sources : United Nations World Population...
44        SP.POP.TOTL                            Data sources : Eurostat
45        SP.POP.GROW                            Data sources : Eurostat
46        SP.POP.GROW  Data sources: United Nations World Population ...
47        SP.POP.TOTL  Data sources : United Nations World Population...
48        SP.POP.TOTL  Data sources : United Nations World Population...
49        SP.POP.GROW  Data sources: United Nations World Population ...
50        SP.POP.GROW  Data sources: United Nations World Population ...
51        SP.POP.TOTL  Data sources : United Nations World Population...
52        SP.POP.TOTL                            Data sources : Eurostat
53        SP.POP.GROW                            Data sources : Eurostat
54        SP.POP.GROW  Data sources: United Nations World Population ...
55        SP.POP.TOTL  Data sources : United Nations World Population...
56        SP.POP.TOTL  Data sources : United Nations World Population...
57        SP.POP.GROW  Data sources: United Nations World Population ...
58        SP.POP.GROW  Data sources: United Nations World Population ...
59        SP.POP.TOTL  Data sources : United Nations World Population...
60        SP.POP.TOTL  Data sources : National Statistical Committee ...
61        SP.POP.GROW  Data sources: National Statistical Committee o...
62        SP.POP.GROW  Data sources: United Nations World Population ...
63        SP.POP.TOTL  Data sources : United Nations World Population...
64        SP.POP.TOTL  Data sources : Bermuda Government Department o...
65        SP.POP.GROW  Data sources: Bermuda Government Department of...
66        SP.POP.GROW  Data sources: United Nations World Population ...
67        SP.POP.TOTL  Data sources : United Nations World Population...
68        SP.POP.TOTL  Data sources : United Nations World Population...
69        SP.POP.GROW  Data sources: United Nations World Population ...
70        SP.POP.GROW  Data sources: United Nations World Population ...
71        SP.POP.TOTL  Data sources : United Nations World Population...
72        SP.POP.TOTL  Data sources : United Nations World Population...
73        SP.POP.GROW  Data sources: United Nations World Population ...
74        SP.POP.GROW  Data sources: United Nations World Population ...
75        SP.POP.TOTL  Data sources : United Nations World Population...
76        SP.POP.TOTL  Data sources : United Nations World Population...
77        SP.POP.GROW  Data sources: United Nations World Population ...
78        SP.POP.GROW  Data sources: United Nations World Population ...
79        SP.POP.TOTL  Data sources : United Nations World Population...
80        SP.POP.TOTL                   Data sources : Statistics Canada
81        SP.POP.GROW                    Data sources: Statistics Canada
82        SP.POP.GROW                            Data sources : Eurostat
83        SP.POP.TOTL                            Data sources : Eurostat
84        SP.POP.TOTL  Data sources : United Nations World Population...
85  SP.POP.0014.TO.ZS                    Refers to Guernsey, and Jersey.
86  SP.POP.1564.TO.ZS                    Refers to Guernsey, and Jersey.
87        SP.POP.GROW  Data sources: United Nations World Population ...
88  SP.POP.TOTL.FE.ZS                    Refers to Guernsey, and Jersey.
89  SP.POP.TOTL.MA.ZS                    Refers to Guernsey, and Jersey.
90        SP.POP.GROW  Data sources: United Nations World Population ...
91        SP.POP.TOTL  Data sources : United Nations World Population...
92        SP.POP.TOTL       Data sources : National Bureau of Statistics
93  SP.POP.0014.TO.ZS        Excluding Hong Kong SAR, Macao SAR, Taiw

## One thing noticed that first two letters of series code and description column have certain relation but there are few deviations also.

In [79]:
country.head(2)

Country Code   Short Name  ... Latest water withdrawal data Unnamed: 31
0          ABW        Aruba  ...                          NaN         NaN
1          AFG  Afghanistan  ...                         2000         NaN

[2 rows x 32 columns]

In [80]:
country['Short Name'].iloc[0:30]

0                      Aruba
1                Afghanistan
2                     Angola
3                    Albania
4                    Andorra
5                 Arab World
6       United Arab Emirates
7                  Argentina
8                    Armenia
9             American Samoa
10       Antigua and Barbuda
11                 Australia
12                   Austria
13                Azerbaijan
14                   Burundi
15                   Belgium
16                     Benin
17              Burkina Faso
18                Bangladesh
19                  Bulgaria
20                   Bahrain
21               The Bahamas
22    Bosnia and Herzegovina
23                   Belarus
24                    Belize
25                   Bermuda
26                   Bolivia
27                    Brazil
28                  Barbados
29                    Brunei
Name: Short Name, dtype: object

In [81]:
country['Table Name'].iloc[30:60]

30                                     Bhutan
31                                   Botswana
32                   Central African Republic
33                                     Canada
34                                Switzerland
35                            Channel Islands
36                                      Chile
37                                      China
38                              Côte d'Ivoire
39                                   Cameroon
40                           Congo, Dem. Rep.
41                                Congo, Rep.
42                                   Colombia
43                                    Comoros
44                                 Cabo Verde
45                                 Costa Rica
46                                       Cuba
47                                    Curaçao
48                             Cayman Islands
49                                     Cyprus
50                             Czech Republic
51                                

In [82]:
country.columns

Index(['Country Code', 'Short Name', 'Table Name', 'Long Name', '2-alpha code',
       'Currency Unit', 'Special Notes', 'Region', 'Income Group', 'WB-2 code',
       'National accounts base year', 'National accounts reference year',
       'SNA price valuation', 'Lending category', 'Other groups',
       'System of National Accounts', 'Alternative conversion factor',
       'PPP survey year', 'Balance of Payments Manual in use',
       'External debt Reporting status', 'System of trade',
       'Government Accounting concept', 'IMF data dissemination standard',
       'Latest population census', 'Latest household survey',
       'Source of most recent Income and expenditure data',
       'Vital registration complete', 'Latest agricultural census',
       'Latest industrial data', 'Latest trade data',
       'Latest water withdrawal data', 'Unnamed: 31'],
      dtype='object')

It's sufficient to recognise a country with their short name, so need to keep columns like Table Name,Long Name

In [83]:
country.drop(['Table Name','Long Name'],axis=1,inplace=True)

In [84]:
country.head()

Country Code   Short Name  ... Latest water withdrawal data Unnamed: 31
0          ABW        Aruba  ...                          NaN         NaN
1          AFG  Afghanistan  ...                         2000         NaN
2          AGO       Angola  ...                         2005         NaN
3          ALB      Albania  ...                         2006         NaN
4          AND      Andorra  ...                          NaN         NaN

[5 rows x 30 columns]

In [85]:
country[country['2-alpha code']==country['WB-2 code']]

Country Code    Short Name  ... Latest water withdrawal data Unnamed: 31
0            ABW         Aruba  ...                          NaN         NaN
1            AFG   Afghanistan  ...                         2000         NaN
2            AGO        Angola  ...                         2005         NaN
3            ALB       Albania  ...                         2006         NaN
4            AND       Andorra  ...                          NaN         NaN
..           ...           ...  ...                          ...         ...
234          WLD         World  ...                          NaN         NaN
235          WSM         Samoa  ...                          NaN         NaN
238          ZAF  South Africa  ...                         2000         NaN
239          ZMB        Zambia  ...                         2002         NaN
240          ZWE      Zimbabwe  ...                         2002         NaN

[233 rows x 30 columns]

In [86]:
country[country['2-alpha code']!=country['WB-2 code']].shape

(8, 30)

from above observation it's clear that 233 rows are same for both so keeping only one column could help us and rows having both the column same value

In [87]:
index=country[country['2-alpha code']!=country['WB-2 code']].index

In [88]:
# Deleting rows having different value for columns 2-alpha code & WB-2 code
country.drop(index,inplace=True)

In [89]:
country.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 233 entries, 0 to 240
Data columns (total 30 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       233 non-null    object 
 1   Short Name                                         233 non-null    object 
 2   2-alpha code                                       233 non-null    object 
 3   Currency Unit                                      207 non-null    object 
 4   Special Notes                                      139 non-null    object 
 5   Region                                             206 non-null    object 
 6   Income Group                                       206 non-null    object 
 7   WB-2 code                                          233 non-null    object 
 8   National accounts base year                        197 non-null    object 
 9   National a

In [90]:
# Remove columns name WB code as it's a clone of 2-alpha code an remove column Unnamed: 31 as it 0 nun null value
country.drop(['WB-2 code','Unnamed: 31'],axis=1,inplace=True)

In [91]:
country.head()

Country Code   Short Name  ... Latest trade data Latest water withdrawal data
0          ABW        Aruba  ...            2012.0                          NaN
1          AFG  Afghanistan  ...            2012.0                         2000
2          AGO       Angola  ...               NaN                         2005
3          ALB      Albania  ...            2012.0                         2006
4          AND      Andorra  ...            2006.0                          NaN

[5 rows x 28 columns]

In [92]:
country.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 233 entries, 0 to 240
Data columns (total 28 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       233 non-null    object 
 1   Short Name                                         233 non-null    object 
 2   2-alpha code                                       233 non-null    object 
 3   Currency Unit                                      207 non-null    object 
 4   Special Notes                                      139 non-null    object 
 5   Region                                             206 non-null    object 
 6   Income Group                                       206 non-null    object 
 7   National accounts base year                        197 non-null    object 
 8   National accounts reference year                   30 non-null     float64
 9   SNA price 

In [93]:
country.iloc[:50]

Country Code  ... Latest water withdrawal data
0           ABW  ...                          NaN
1           AFG  ...                         2000
2           AGO  ...                         2005
3           ALB  ...                         2006
4           AND  ...                          NaN
5           ARB  ...                          NaN
6           ARE  ...                         2005
7           ARG  ...                         2000
8           ARM  ...                         2007
9           ASM  ...                          NaN
10          ATG  ...                         2005
11          AUS  ...                         2000
12          AUT  ...                         2002
13          AZE  ...                         2005
14          BDI  ...                         2000
15          BEL  ...                         2007
16          BEN  ...                         2001
17          BFA  ...                         2001
18          BGD  ...                         2008
19          BGR  ...                         2009
20          BHR  ...                         2003
21          BHS  ...                          NaN
22          BIH  ...                         2009
23          BLR  ...                         2000
24          BLZ  ...                         2000
25          BMU  ...                          NaN
26          BOL  ...                         2000
27          BRA  ...                         2006
28          BRB  ...                         2005
29          BRN  ...                         1994
30          BTN  ...                         2008
31          BWA  ...                         2000
32          CAF  ...                         2005
33          CAN  ...                         1986
34          CHE  ...                         2000
36          CHL  ...                         2007
37          CHN  ...                         2005
38          CIV  ...                         2005
39          CMR  ...                         2000
41          COG  ...                         2002
42          COL  ...                         2000
43          COM  ...                         1999
44          CPV  ...                         2001
45          CRI  ...                         1997
46          CUB  ...                         2007
47          CUW  ...                          NaN
48          CYM  ...                          NaN
49          CYP  ...                         2009
50          CZE  ...                         2007
51          DEU  ...                         2007

[50 rows x 28 columns]

In [104]:
country[~country['Latest industrial data'].isnull()]

Country Code     Short Name  ... Latest trade data Latest water withdrawal data
3            ALB        Albania  ...            2012.0                         2006
7            ARG      Argentina  ...            2012.0                         2000
11           AUS      Australia  ...            2012.0                         2000
12           AUT        Austria  ...            2012.0                         2002
13           AZE     Azerbaijan  ...            2012.0                         2005
..           ...            ...  ...               ...                          ...
223          UGA         Uganda  ...            2012.0                         2002
226          URY        Uruguay  ...            2012.0                         2000
227          USA  United States  ...            2012.0                         2005
232          VNM        Vietnam  ...            2011.0                         2005
238          ZAF   South Africa  ...            2012.0                         2000

[104 rows x 28 columns]

for now let's delete

In [105]:
stats_series.head()

Series Code       Topic  ... License Type Unnamed: 20
0  BAR.NOED.1519.FE.ZS  Attainment  ...          NaN         NaN
1     BAR.NOED.1519.ZS  Attainment  ...          NaN         NaN
2  BAR.NOED.15UP.FE.ZS  Attainment  ...          NaN         NaN
3     BAR.NOED.15UP.ZS  Attainment  ...          NaN         NaN
4  BAR.NOED.2024.FE.ZS  Attainment  ...          NaN         NaN

[5 rows x 21 columns]

In [108]:
stats_series['Indicator Name'].unique().shape

(3665,)

In [114]:
stats_series['Indicator Name'][9]

'Barro-Lee: Percentage of population age 25+ with no education'

In [110]:
stats_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null     object 
 11  Notes from original source    

In [115]:
stats_data.head()

Country Name Country Code  ... 2100 Unnamed: 69
0   Arab World          ARB  ...  NaN         NaN
1   Arab World          ARB  ...  NaN         NaN
2   Arab World          ARB  ...  NaN         NaN
3   Arab World          ARB  ...  NaN         NaN
4   Arab World          ARB  ...  NaN         NaN

[5 rows x 70 columns]

In [119]:
stats_data.shape

(886930, 70)

In [118]:
stats_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

## We are storing only the years data from 2005-2017 as after that all are projections

In [122]:
required_columns=['Country Name','Country Code','Indicator Name','Indicator Code','2005','2006','2007','2008','2009','2010','2011','2012','2013','2014','2014','2015','2016','2017']

In [125]:
# We are updating our data with required columns only
stats_data=stats_data[required_columns]

In [126]:
stats_data.head()

Country Name Country Code  ... 2016 2017
0   Arab World          ARB  ...  NaN  NaN
1   Arab World          ARB  ...  NaN  NaN
2   Arab World          ARB  ...  NaN  NaN
3   Arab World          ARB  ...  NaN  NaN
4   Arab World          ARB  ...  NaN  NaN

[5 rows x 18 columns]

In [128]:
# Let's check some insights for India
stats_data.loc[stats_data['Country Name']=='India']

Country Name Country Code  ... 2016 2017
414145        India          IND  ...  NaN  NaN
414146        India          IND  ...  NaN  NaN
414147        India          IND  ...  NaN  NaN
414148        India          IND  ...  NaN  NaN
414149        India          IND  ...  NaN  NaN
...             ...          ...  ...  ...  ...
417805        India          IND  ...  NaN  NaN
417806        India          IND  ...  NaN  NaN
417807        India          IND  ...  NaN  NaN
417808        India          IND  ...  NaN  NaN
417809        India          IND  ...  NaN  NaN

[3665 rows x 18 columns]

In [129]:
stats_data.loc[stats_data['Country Name']=='India'].shape

(3665, 18)

In [135]:
# Lets store this dataframe called India
India=stats_data.loc[stats_data['Country Name']=='India'].reset_index(drop=True)

In [136]:
India

Country Name Country Code  ... 2016 2017
0           India          IND  ...  NaN  NaN
1           India          IND  ...  NaN  NaN
2           India          IND  ...  NaN  NaN
3           India          IND  ...  NaN  NaN
4           India          IND  ...  NaN  NaN
...           ...          ...  ...  ...  ...
3660        India          IND  ...  NaN  NaN
3661        India          IND  ...  NaN  NaN
3662        India          IND  ...  NaN  NaN
3663        India          IND  ...  NaN  NaN
3664        India          IND  ...  NaN  NaN

[3665 rows x 18 columns]

In [139]:
India.shape

(3665, 18)

In [141]:
India['Indicator Name'].unique().shape

(3665,)

#Checking indicator column for dataframe India what this column really contains

In [137]:
India['Indicator Name'][0]

'Adjusted net enrolment rate, lower secondary, both sexes (%)'

In [138]:
India['Indicator Name'][1]

'Adjusted net enrolment rate, lower secondary, female (%)'

In [143]:
India['Indicator Name'][2]

'Adjusted net enrolment rate, lower secondary, gender parity index (GPI)'

In [144]:
India['Indicator Name'][3]

'Adjusted net enrolment rate, lower secondary, male (%)'

In [145]:
India['Indicator Name'][4]

'Adjusted net enrolment rate, primary, both sexes (%)'

In [146]:
India['Indicator Name'][5]

'Adjusted net enrolment rate, primary, female (%)'

In [147]:
India['Indicator Name'][6]

'Adjusted net enrolment rate, primary, gender parity index (GPI)'

In [148]:
India['Indicator Name'][7]

'Adjusted net enrolment rate, primary, male (%)'

In [160]:
list(India['Indicator Name'].iloc[:100])

['Adjusted net enrolment rate, lower secondary, both sexes (%)',
 'Adjusted net enrolment rate, lower secondary, female (%)',
 'Adjusted net enrolment rate, lower secondary, gender parity index (GPI)',
 'Adjusted net enrolment rate, lower secondary, male (%)',
 'Adjusted net enrolment rate, primary, both sexes (%)',
 'Adjusted net enrolment rate, primary, female (%)',
 'Adjusted net enrolment rate, primary, gender parity index (GPI)',
 'Adjusted net enrolment rate, primary, male (%)',
 'Adjusted net enrolment rate, upper secondary, both sexes (%)',
 'Adjusted net enrolment rate, upper secondary, female (%)',
 'Adjusted net enrolment rate, upper secondary, gender parity index (GPI)',
 'Adjusted net enrolment rate, upper secondary, male (%)',
 'Adjusted net intake rate to Grade 1 of primary education, both sexes (%)',
 'Adjusted net intake rate to Grade 1 of primary education, female (%)',
 'Adjusted net intake rate to Grade 1 of primary education, gender parity index (GPI)',
 'Adjusted 

In [159]:
India[~India.isnull()]

Country Name Country Code  ... 2016 2017
0           India          IND  ...  NaN  NaN
1           India          IND  ...  NaN  NaN
2           India          IND  ...  NaN  NaN
3           India          IND  ...  NaN  NaN
4           India          IND  ...  NaN  NaN
...           ...          ...  ...  ...  ...
3660        India          IND  ...  NaN  NaN
3661        India          IND  ...  NaN  NaN
3662        India          IND  ...  NaN  NaN
3663        India          IND  ...  NaN  NaN
3664        India          IND  ...  NaN  NaN

[3665 rows x 18 columns]

In [130]:
stats_data.loc[stats_data['Country Name']=='India','Indicator Name'].unique()

array(['Adjusted net enrolment rate, lower secondary, both sexes (%)',
       'Adjusted net enrolment rate, lower secondary, female (%)',
       'Adjusted net enrolment rate, lower secondary, gender parity index (GPI)',
       ..., 'Youth literacy rate, population 15-24 years, female (%)',
       'Youth literacy rate, population 15-24 years, gender parity index (GPI)',
       'Youth literacy rate, population 15-24 years, male (%)'],
      dtype=object)

In [95]:
country.shape

(233, 28)